api/handlers/user_handler.go

```diff
@@ -0,0 +1,167 @@
+package handlers
+
+import (
+	"errors"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/odpf/columbus/star"
+	"github.com/odpf/columbus/user"
+	"github.com/odpf/salt/log"
+)
+
+// UserHandler exposes a REST interface to user
+type UserHandler struct {
+	starRepository star.Repository
+	logger         log.Logger
+}
+
+func (h *UserHandler) GetStarredAssetsWithHeader(w http.ResponseWriter, r *http.Request) {
+	userID := user.FromContext(r.Context())
+	if userID == "" {
+		h.logger.Warn(errMissingUserID.Error())
+		WriteJSONError(w, http.StatusBadRequest, errMissingUserID.Error())
+		return
+	}
+
+	starCfg := buildStarConfig(h.logger, r.URL.Query())
+
+	starredAssets, err := h.starRepository.GetAllAssetsByUserID(r.Context(), starCfg, userID)
+	if err != nil {
+		if errors.Is(err, star.ErrEmptyUserID) {
+			WriteJSONError(w, http.StatusBadRequest, err.Error())
+			return
+		}
+		if errors.As(err, new(star.NotFoundError)) {
+			WriteJSONError(w, http.StatusNotFound, err.Error())
+			return
+		}
+		internalServerError(w, h.logger, err.Error())
+		return
+	}
```

Comment:
Should we refactor this line to a function as it is also being used by `GetStarredAssetsWithPath` method?

Reply:
We could but I think we don't need to, unless it is being used in most places. Making this as it is so it is more readable and make the entry and exit point clearer.